R/by_aesthetics.R

```diff
@@ -123,7 +123,7 @@ by_col = function(ngrps = 1L, col = NULL, palette = NULL, gradient = NULL, order
 
 by_pch = function(ngrps, type, pch = NULL) {
   no_pch = FALSE
-  if (!type %in% c("p", "b", "o", "pointrange", "errorbar", "boxplot", "qq")) {
+  if (!type %in% c("p", "b", "o", "pointrange", "errorbar", "boxplot", "qq", "text")) {
```

Comment:
Interesting idea. I thought about it a bit, but I'm not convinced this is a good idea. I think say you draw: `mpg ~ hp | factor(cyl)`, then you'll get a legend that says "a 4", "b 6", "c 8". I think that's just confusing. Inattentive readers will wonder if there's an "a4" category.

I do agree that the default circle isn't great for this, so I changed the default to 15. That seems neutral and nice, and maybe should be the default for our non-point legends. What do you think?

![text_facets](https://github.com/user-attachments/assets/423a4ae3-7791-491e-81a2-3b0bda6ea109)

Reply:
Hmmm. It's something to chew over. I know that ggplot2 takes the "a" approach for labeling grouped colouring, for example. https://ggplot2.tidyverse.org/reference/geom_text.html#ref-examples

My ideal would probably be "abc" but (a) I don't know how to do this with pch and (b) not everyone might share my preferences.

Let me think on it for a bit.

Reply:
Right. I do think that the `ggplot2` has less potential for confusion, since the letter stays constant and clearly refers to the geom.

Reply:
> Right. I do think that the ggplot2 has less potential for confusion, since the letter stays constant and clearly refers to the geom.

But we could ensure that letter stays constant too with `pch = 97` (i.e., no sequencing)? That's actually what my initial suggestion was, although I phrased it badly :-/

(Sequencing was just meant to be an option.)

Reply:
> clearly refers to the geom.

To clarify: Do you mean the fact that the "a" legend symbol also inherits the grey theme background?

Reply:
No, that was just an argument against sequencing, a, b, c, etc.

I'm fine with a fixed `pch=97` if that's what you prefer, but then it might be nice to have a box around it, so it doesn't look like "a4". Not sure how to do that.

Reply:
Got it, thanks. There might be a way to do this with our existing `draw_legend.R` logic. I need to head out for a bit, but will think on this and then get back to you.

Reply:
Do you mind trying the following and let me know what you think?

- First, impose `pch = 97` to give "a" legend labels.
- Second, in draw_legend.R, immediately after l. 454 here https://github.com/vincentarelbundock/tinyplot/blob/1ca5579bd6c3a142912ad082a2bcd6c1a61ac59e/R/draw_legend.R#L450-L454, add the following:

```r
if (type == "text") {
  recordGraphics(
    do.call("legend", legend_args),
    list(legend_args = modifyList(
      legend_args),
      pch = 22, pt.cex = 3, pt.lwd = 0.5,
      col = par("fg"), bg = "transparent", text.col = "transparent",
      keep.null = TRUE
      ),
    getNamespace("tinyplot")
  )
}
```

The idea is that we add thin boxes around the "a" pch labels, but keep (most of) the remaining legend args the same. 

Again, I'm not saying we _have_ to do this. But curious to see if it looks good. I'm happy for you to make the final call...

Reply:
Here's the side-by-side. What do you think?

![text_facets](https://github.com/user-attachments/assets/c5742ab8-84e5-42a8-89ea-3f4e6fc50079)

Reply:
Wife says: "it feels like the 'a' should give me some information".

(Disclosure: I might not have quoted this if she disagreed with me 🤪)

Reply:
Hmmm, okay let's just stick with the filled squares. It's less complicated and the alpha version isn't obviously better.

(I could try again down the line by improving the spacing between the points and text, much like we do for area legends. But for now let's keep it as-is.)

Sorry for the back and forth!

Am I correct in thinking that the current PR can be merged then? You haven't pushed these experimental changes, right?

Reply:
That makes sense. I'm a big fan of iteration and gradual improvement. None of these decisions are crucial, and we can always revisit later.

Yep, this should be OK to merge.